simulator_main/services/BitstreamService.cs

```diff
@@ -9,20 +12,30 @@ namespace simulator_main.services
 {
     public class BitstreamService : IBitstreamService
     {
-        private Dictionary<int, string> IcdDictionary;
+        private Dictionary<int, (Type,string)> IcdDictionary;
```

Comment:
Chnage that key of dictionary that will be name of icd and value will be the type